admin/model_serverless_metric_threshold.go

```diff
@@ -4,354 +4,227 @@ package admin
+	if o == nil || IsNil(o.MetricName) {
+		var ret string
+		return ret
 	}
+	return *o.MetricName
 }
 
-// TimeMetricThresholdAsServerlessMetricThreshold is a convenience function that returns TimeMetricThreshold wrapped in ServerlessMetricThreshold
-func TimeMetricThresholdAsServerlessMetricThreshold(v *TimeMetricThreshold) ServerlessMetricThreshold {
-	return ServerlessMetricThreshold{
-		TimeMetricThreshold: v,
+// GetMetricNameOk returns a tuple with the MetricName field value if set, nil otherwise
+// and a boolean to check if the value has been set.
+func (o *ServerlessMetricThreshold) GetMetricNameOk() (*string, bool) {
+	if o == nil || IsNil(o.MetricName) {
+		return nil, false
 	}
+	return o.MetricName, true
 }
 
-// Unmarshal JSON data into one of the pointers in the struct
-func (dst *ServerlessMetricThreshold) UnmarshalJSON(data []byte) error {
-	var err error
-	// use discriminator value to speed up the lookup
-	var jsonDict map[string]interface{}
-	err = newStrictDecoder(data).Decode(&jsonDict)
-	if err != nil {
-		return fmt.Errorf("failed to unmarshal JSON into map for the discriminator lookup")
-	}
-
-	// check if the discriminator value is 'DataMetricThreshold'
-	if jsonDict["metricName"] == "DataMetricThreshold" {
-		// try to unmarshal JSON data into DataMetricThreshold
-		err = json.Unmarshal(data, &dst.DataMetricThreshold)
-		if err == nil {
-			return nil // data stored in dst.DataMetricThreshold, return on the first match
-		} else {
-			dst.DataMetricThreshold = nil
-			return fmt.Errorf("failed to unmarshal ServerlessMetricThreshold as DataMetricThreshold: %s", err.Error())
-		}
+// HasMetricName returns a boolean if a field has been set.
+func (o *ServerlessMetricThreshold) HasMetricName() bool {
+	if o != nil && !IsNil(o.MetricName) {
+		return true
 	}
 
-	// check if the discriminator value is 'RPUMetricThreshold'
-	if jsonDict["metricName"] == "RPUMetricThreshold" {
-		// try to unmarshal JSON data into RPUMetricThreshold
-		err = json.Unmarshal(data, &dst.RPUMetricThreshold)
-		if err == nil {
-			return nil // data stored in dst.RPUMetricThreshold, return on the first match
-		} else {
-			dst.RPUMetricThreshold = nil
-			return fmt.Errorf("failed to unmarshal ServerlessMetricThreshold as RPUMetricThreshold: %s", err.Error())
-		}
-	}
-
-	// check if the discriminator value is 'RawMetricThreshold'
-	if jsonDict["metricName"] == "RawMetricThreshold" {
-		// try to unmarshal JSON data into RawMetricThreshold
-		err = json.Unmarshal(data, &dst.RawMetricThreshold)
-		if err == nil {
-			return nil // data stored in dst.RawMetricThreshold, return on the first match
-		} else {
-			dst.RawMetricThreshold = nil
-			return fmt.Errorf("failed to unmarshal ServerlessMetricThreshold as RawMetricThreshold: %s", err.Error())
-		}
-	}
+	return false
+}
 
-	// check if the discriminator value is 'SERVERLESS_AVG_COMMAND_EXECUTION_TIME'
-	if jsonDict["metricName"] == "SERVERLESS_AVG_COMMAND_EXECUTION_TIME" {
```

Comment:
Those values seem invalid (capitalized). Discriminator here is redundant as all fields are the same).
With so many types client side logic is hard to support